src/MiniProfiler.Shared/ui/lib/MiniProfiler.ts

```diff
@@ -886,125 +899,302 @@ namespace StackExchange.Profiling {
                 return;
             }
 
-            const result = this.renderProfiler(json).addClass('new');
+            const profilerHtml = this.renderProfiler(json, true);
 
             if (this.controls) {
-                result.insertBefore(this.controls);
+                this.controls.insertAdjacentHTML('beforebegin', profilerHtml);
             } else {
-                result.appendTo(this.container);
+                this.container.insertAdjacentHTML('beforeend', profilerHtml);
             }
 
             // limit count to maxTracesToShow, remove those before it
-            this.container.find('.mp-result:lt(' + -this.options.maxTracesToShow + ')').remove();
+            const results = this.container.querySelectorAll('.mp-result');
+            const toRemove = results.length - this.options.maxTracesToShow;
+            for (let i = 0; i < toRemove; i++) {
+                results[i].parentNode.removeChild(results[i]);
+            }
         }
 
-        private scrollToQuery = (link: JQuery, queries: JQuery, whatToScroll: JQuery) => {
-            const id = link.closest('tr').data('timing-id');
-            const rows = queries.find('tr[data-timing-id="' + id + '"]').addClass('highlight');
-
-            // ensure they're in view
-            whatToScroll.scrollTop(whatToScroll.scrollTop() + rows.position().top - 100);
+        private scrollToQuery = (link: HTMLElement, queries: HTMLElement, whatToScroll: HTMLElement) => {
+            const id = link.closest('tr').dataset['timingId'];
+            const rows = queries.querySelectorAll('tr[data-timing-id="' + id + '"]');
+            rows.forEach(n => n.classList.add('highlight'));
+            const first = rows && rows[0];
+            if (first) {
+                // ensure they're in view
+                whatToScroll.scrollTop = whatToScroll.scrollTop + first.getBoundingClientRect().top - 100;
```

Comment:
You can look into https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView

Reply:
Oh awesome, TIL - thanks! I'll test this out tonight.

Reply:
Done! https://github.com/MiniProfiler/dotnet/pull/442/commits/49aa25b5193a417d5048e1fef6d22fba246e3503